keras/src/callbacks/orbax_checkpoint.py

```
@@ -0,0 +1,579 @@
+        self.manager.wait_until_finished()
+        if self.verbose > 0:
+            print("OrbaxCheckpoint: All saves finalized.")
+
+    def load_checkpoint(self, step, model=None):
+        """Load model and optimizer state from a specific checkpoint step.
+
+        Args:
+            step: The checkpoint step to load from.
+            model: Optional model to load into. If None, loads into self.model.
+
+        Returns:
+            tuple: (success, iterator_state) where success is True if loading
+            was successful, False otherwise, and iterator_state is the saved
+            data iterator state dict if available, None otherwise.
+        """
+        # In distributed training, only load on primary process
+        if backend.get_process_index() != 0:
+            return True  # Return True to indicate no error, but no loading
+            # performed
+
+        try:
+            if self.verbose > 0:
+                print(
+                    f"OrbaxCheckpoint: Loading checkpoint from step {step}..."
+                )
+
+            # Prepare restore arguments - Orbax can restore without explicit
+            # template
+            restore_args = ocp.args.StandardRestore()
+
+            # Load the checkpoint
+            checkpoint_data = self.manager.restore(step, args=restore_args)
+
+            # Restore the model state
+            target_model = model if model is not None else self.model
+            success = self._restore_model_state(checkpoint_data, target_model)
+
+            # Extract iterator state if available
+            iterator_state = checkpoint_data.get("data_iterator", None)
+
+            return success, iterator_state
+
+        except Exception as e:
+            if self.verbose > 0:
+                print(
+                    f"OrbaxCheckpoint: Failed to load checkpoint from step "
+                    f"{step}: {e}"
+                )
+            return False, None
```

I think we should remove the try / except and let the error go through.

If the checkpoint fails to load, this accidentally trains the model from scratch instead of continuing training. Users won't realize.

Or am I missing something?